lib/orwell/web/router.ex

```diff
@@ -28,6 +28,8 @@ defmodule Orwell.Web.Router do
     delete "/auth", AuthController, :delete
     get "/auth/:provider", AuthController, :request
     get "/auth/:provider/callback", AuthController, :callback
+
+    get "/*path", PostController, :show
```

Comment:
I don't see why making 2 requests is such a big deal. You just need to fetch the posts list like you're already doing for the index, look for the one with a matching ID, and get that file. No need to, as you seem to suggest, "collect every file from the git repository"

also, we can probably cache the first request later on. We can use a simple GenServer for that.

I honestly prefer this much more than having the URL become coupled to the git repo's representation of our blog, and prone to possible exploits